### ISSUE TITLE
Problems with CSS

### ISSUE THREAD

**Issue by osterchrisi** [victoriadrake/hugo-theme-introduction#227]
I am having trouble getting the page to build. I always get the following error:

```
Building sites … /home/chrisi/Homepage/greeenparts.com/node_modules/fs-extra/lib/mkdirs/make-dir.js:85
      } catch {
              ^

SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:617:28)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/chrisi/Homepage/greeenparts.com/node_modules/fs-extra/lib/mkdirs/index.js:3:44)
Total in 326 ms
Error: Error building site: POSTCSS: failed to transform "css/main.css" (text/css): /home/chrisi/Homepage/greeenparts.com/node_modules/fs-extra/lib/mkdirs/make-dir.js:85
      } catch {
              ^

SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:617:28)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/chrisi/Homepage/greeenparts.com/node_modules/fs-extra/lib/mkdirs/index.js:3:44)

```
I absolutely feel it has something to do wint the postcss-cli and autoprefixer modules but also I think I have them installed locally (in the page directory) in the correct version. I am struggling to install them globally in the correct version though. This is me locally as the building user listing the modules:

`/home/chrisi/Homepage/greeenparts.com
└── postcss-cli@7.1.1 `
`/home/chrisi/Homepage/greeenparts.com
└── autoprefixer@9.8.5`

I love this theme and would be so grateful if someone could help me out. I have lost of experience with Unix systems and have built another homepage with Hugo and a different theme already but I am pretty stuck when it comes to these node.js issues.

```
chrisi@mobstation ~/Homepage/greeenparts.com:  node -v 
v10.15.1
 chrisi@mobstation ~/Homepage/greeenparts.com:  npm -v
6.4.1
 chrisi@mobstation ~/Homepage/greeenparts.com:  nvm --version
0.35.3

```
Thank you for any help on this!!

**Comment by victoriadrake** [comment]
Hi there! Could you explain what you mean about "struggling to install them globally in the correct version"? If you run `npm install -g autoprefixer postcss-cli` to install these packages globally, what happens?

**Comment by osterchrisi** [comment]
Thanks for your input, this might have nudged me towards the correct direction already. I ran your command via _sudo_ before and therefor running it as my "normal" user gave me:
```
npm install -g autoprefixer postcss-cli
npm WARN checkPermissions Missing write access to /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/node_modules/caniuse-lite
npm WARN checkPermissions Missing write access to /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/node_modules/electron-to-chromium
npm WARN checkPermissions Missing write access to /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer
npm WARN checkPermissions Missing write access to /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/postcss-cli
npm WARN checkPermissions Missing write access to /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/node_modules
npm WARN checkPermissions Missing write access to /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/postcss-cli/node_modules
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.1.2 (node_modules/postcss-cli/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! path /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/node_modules/caniuse-lite
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access
npm ERR! Error: EACCES: permission denied, access '/home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/node_modules/caniuse-lite'
npm ERR!  { [Error: EACCES: permission denied, access '/home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/node_modules/caniuse-lite']
npm ERR!   stack:
npm ERR!    'Error: EACCES: permission denied, access \'/home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/node_modules/caniuse-lite\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path:
npm ERR!    '/home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/node_modules/caniuse-lite' }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator (though this is not recommended).

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chrisi/.npm/_logs/2020-08-02T15_03_07_965Z-debug.log

```
I checked and this was due to the fact that the _autoprefixer_ directory and everything below (same is true for the postcss-cli directory) was owned by my root user - makes sense since it was that user that installed the modules. I nvm removed the modules (autoprefixer and postcss-cli) globally as root again and then re-run your command as "normal" usesr:
```
npm install -g autoprefixer postcss-cli
/home/chrisi/.nvm/versions/node/v10.15.1/bin/autoprefixer -> /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/autoprefixer/bin/autoprefixer
/home/chrisi/.nvm/versions/node/v10.15.1/bin/postcss -> /home/chrisi/.nvm/versions/node/v10.15.1/lib/node_modules/postcss-cli/bin/postcss
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.1.2 (node_modules/postcss-cli/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

+ postcss-cli@7.1.1
+ autoprefixer@9.8.6
added 133 packages from 108 contributors in 5.652s
```
So that seems to work. But I still get the same build error as in my top post when running _hugo_. Is it necessary to get the optional dependencies going?
Thank you so much for your help, this is a beautiful theme!!

**Comment by victoriadrake** [comment]
It may be necessary to get the optional dependencies. I haven't seen this behavior before with the global installation, so I regret I can't tell you for sure.

In the meantime I'm leaving this one open for help wanted in case others have more to contribute!

**Comment by osterchrisi** [comment]
Okay, too bad! The optional dependency seems to be an issue with my operating system, so I think I won't be able to change that :P But thank you anyway for your efforts Victoria, I appreciate it. If anyone else can help here, thank you too!

**Comment by SOERGI** [comment]
> Okay, too bad! The optional dependency seems to be an issue with my operating system, so I think I won't be able to change that :P But thank you anyway for your efforts Victoria, I appreciate it. If anyone else can help here, thank you too!

Not sure if people still run into this, but I did and the solution was to update node to a current version. In my case I had a version 8 installed and updated to a 14. Also, I installed the two npm packages globally. 
As to the why: my best guess is that the try-catch construct without the error parameter was not supported pre-node 10, see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)
